esp-hal/src/systimer.rs

```diff
@@ -45,31 +46,93 @@ use crate::{
 
 // TODO this only handles unit0 of the systimer
 
-pub struct SystemTimer<'d> {
```

Comment:
I might be missing something, but couldn't `SystemTimer` have a MODE on it too, then we wouldn't need to do this `create` dance.